src/pages/Product/ProductPage.js

```diff
@@ -16,6 +16,7 @@ const ProductPage = ({ onAddToCart, setCartItems, cartItems }) => {
     if (!cartItems.includes(product)) {
```

Comment:
@sandrina-p and this here is supposed to prevent an item being added to the cart twice since there's only one available. It works until the app re-renders and then it adds an already existing item.

Reply:
This has to do with reference equality in JS, remember? 

```js
var obj1 = { a: 1 }
var list = [obj1, { b: 2}]
obj1 === { a: 1 } // false
obj1 === list[0] // true
```

So, this works for the first time because the reference is exactly the same. But on the page refresh, the reference isn't the same anymore because that object was converted with JSON stringify/parsed.

My suggestion: Only store the `productId` (it's a string) on the local storage and use it to check if it already exists

Reply:
Will do, thanks again!

Reply:
So you mean not store the product object in local storage at all? As far as I understand that would mean I'd need separate states for cart items and id's as well as having to rewrite most of the logic in Cart and ProductPage. I have a feeling it has to be simpler than that but I'm not seeing it 😄

Reply:
Just store de product id and then use that to fetch the entire product from the server again

Reply:
Is there anything wrong with simply replacing 
``` if (!cartItems.includes(product))``` with ``` if (!JSON.parse(localStorage.getItem("item").includes(product.id)))``` ? Works that way too.

Reply:
I don't see how is that working. The `item` is an array of objects. You are looking for the id only, so that would never match. 🤔 

You could use `Array.find()` to check for the id in `cartItems`

Reply:
I honestly have no clue 😆  I'll leave it commented out, I'm curious if it breaks when you run it

Reply:
As I suspected, this isn't working. Are you sure it's working on your side?

On my side, the same item can be added again, after page refresh: [screenshot](https://user-images.githubusercontent.com/14869087/108049355-0e981d80-7040-11eb-8f19-b63a8cc8a4c2.png)

Reply:
I understand now how that worked. Let's look at the code again

```
JSON.parse(localStorage.getItem("item").includes(product.id))
```
First of all, here the `includes` is being executed to the localStorage item directly, not to the result of `JSON.parse`.

`includes` is a method that can be used on both array and strings.

When we do `localStorage.getItem("item")`, it always returns a string. So, the product id will be inside that string, somewhere That's why it works.

```
JSON.parse(localStorage.getItem("item").includes(product.id))

// The code above does the following in the following order:
// gets the items
const item = localStorage.getItem("item") // "[{ productId: '123', name: 'Macrame' }]"

// In the string, look for productId, let's assume it's "123"
const hasProductId = item.includes(productId) // true - because "123" is part of the string.

// Parse the "true" result to "true".
const productParsed = JSON.parse(hasProductId)

// If instead we did what was supposed to be done, this wouldn't work.
// First parse the localstorage to an array and then look for product id inside it.
JSON.parse(localStorage.getItem("item")).includes(product.id)
```

Reply:
This is super helpful, thanks for breaking it down for me! 😄 I see now why this is not a reliable way.